File: easy/3300_minimum_element_after_replacement_with_digit_sum/solution.cpp
#include <vector>

class Solution {
public:
  int minElement(std::vector<int> &nums) {
    int minReplacement = INT_MAX;

    for (int &num : nums) {
      int replacement = 0;

      while (num) {
        replacement += num % 10;
        num /= 10;
      }

      minReplacement = std::min(minReplacement, replacement);
    }

    return minReplacement;
  }
};

File: easy/3300_minimum_element_after_replacement_with_digit_sum/solution.py
class Solution:
    def minElement(self, nums: list[int]) -> int:
        min_replacement = float("inf")

        for num in nums:
            replacement = 0

            while num:
                replacement += num % 10
                num //= 10

            min_replacement = min(min_replacement, replacement)

        return min_replacement
